smoke.lic

```diff
@@ -31,22 +42,29 @@ class Smoker
       smoke_settings = get_settings.smoke
 
       @cigar = args.cigar
-      @image = smoke_settings['smoke_image'] if smoke_settings['smoke_image']
+      if smoke_settings['smoke_image']
+        if smoke_settings['smoke_image'].is_a? Array
+          @image = smoke_settings['smoke_image'].sample
+        else
+          @image = smoke_settings['smoke_image']
+        end
+      end
+      
       @lighter_info = smoke_settings['lighter']
       @cigar_container = smoke_settings['cigar_container']
     end
 
     Flags.add('new-cig', 'That was the last of your')
 
-    smoke_cig if get_cig? && light_cig?
+    smoke_cig if light_cig? if get_cig?
 
     Flags.delete('new-cig')
   end
 
   def get_cig?
-    case bput("get #{@cigar} in my #{@cigar_container}", 'You get', 'What were', 'You need')
+    case DRC.bput("get #{@cigar} in my #{@cigar_container}", 'You get', 'What were', 'You need')
```

Comment:
```suggestion
    Unless DRCI.get_item?(@cigar, @cigar_container)
```

Would allow use of portals, and has the match handling from items. Then just put your comment below, with false, and else, followed by true.